source/builders.txt

```diff
@@ -0,0 +1,305 @@
+- Prints the matching documents
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-find
+      :end-before: end-find
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: json
+      :visible: false
+
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9137115,9.3390503],...}
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9357223,9.3340302],...}
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9081268,9.3547792],...}
+      // Results truncated
+
+To learn more about find operations, see the :ref:`php-retrieve` guide.
+
+Delete Example
+~~~~~~~~~~~~~~
+
+This example performs the following actions:
+
+- Uses the ``Query::regex()`` factory method to match documents in which
+  the ``feature_type`` field value contains the string ``'nondangerous'``
+- Calls the :phpmethod:`MongoDB\Collection::deleteOne()`
+  method to delete the first matching document
+- Prints the number of deleted documents
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-deleteone
+      :end-before: end-deleteone
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Deleted documents: 1
+
+To learn more about delete operations, see the :ref:`php-write-delete`
+guide.
+
+.. _php-builders-update:
+
+Define an Update Document
+-------------------------
+
+You can use factory methods from the ``Stage`` builder class to update
+or create field values in your update operations. To learn more about
+updating documents, see the :ref:`php-write-update` guide.
+
+.. important::
+    
+   The only update operation that you can express by using builders uses
+   the ``$set`` operator. This operator allows you to replace the value of
+   a field with a specified value or create a new field that has a specified
+   value. To learn more, see the :manual:`$set </reference/operator/update/set/>`
+   reference in the {+mdb-server+} manual.
+
+The following steps describe how to create an update document by using
+builders:
+
+1. Create a ``Pipeline`` instance.
+   
+#. Pass one or more ``$set`` stages by calling the ``Stage::set()``
+   method and passing a field name and a value.
+
+The following code shows the template to define an update by using
+builders:
+
+.. code-block:: php
+    
+   $update = new Pipeline(
+       Stage::set(<field name>: <value>),
+       Stage::set(<field name>: <value>),
+       ...
+   );
+
+This example performs the following actions:
+
+- Uses the ``Query::eq()`` factory method to match documents in which
+  the ``watlev`` field value is ``'partly submerged at high water'``
+- Uses builders to specify an update that sets
+  sets the ``year`` field to ``1870``
+- Calls the :phpmethod:`MongoDB\Collection::updateOne()`
+  method to perform the update
+- Prints the number of updated documents
+
+.. io-code-block::
```

Comment:
S: For long code examples like this, I think code comments would be helpful to include at each section break